app/ts/background/background-startup.ts

```diff
@@ -186,6 +186,8 @@ async function startup() {
 		}
 	})
 
+	setInterval(async () => checkIfInterceptorShouldSleep(simulator.ethereum), 1000 )
```

Comment:
an async function in setInterval looks like a race condition, did you mean to run this synchronously?

```suggestion
	setInterval(() => checkIfInterceptorShouldSleep(simulator.ethereum), 1000 )
```